test/e2e/cleanup/submariner.go

```diff
@@ -0,0 +1,264 @@
+		Expect(err).To(Succeed())
+
+		submarinerInterface = operatorClient.SubmarinerV1alpha1().Submariners(framework.TestContext.SubmarinerNamespace)
+
+		submariner, err = submarinerInterface.Get(context.TODO(), submarinercr.SubmarinerName, metav1.GetOptions{})
+		Expect(err).To(Succeed())
+
+		brokerRestConfig = getBrokerRestConfig(submariner.Spec.BrokerK8sRemoteNamespace)
+		Expect(brokerRestConfig).ToNot(BeNil(), "No broker located")
+
+		routeAgentPodMonitor = startPodMonitor(names.AppendUninstall(names.RouteAgentComponent), stopCh)
+		gatewayPodMonitor = startPodMonitor(names.AppendUninstall(names.GatewayComponent), stopCh)
+	})
+
+	AfterEach(func() {
+		if submariner != nil {
+			submariner.ObjectMeta = metav1.ObjectMeta{
+				Name:        submariner.Name,
+				Labels:      submariner.Labels,
+				Annotations: submariner.Annotations,
+			}
+
+			_, err := submarinerInterface.Create(context.TODO(), submariner, metav1.CreateOptions{})
+			if err != nil {
+				framework.Errorf("Error re-creating Submariner: %v", err)
+			}
+
+			By("Re-created Submariner resource")
+		}
+	})
+
+	It("should run DaemonSets/Deployments to uninstall components", func() {
+		By("Deleting Submariner resource")
+
+		Expect(submarinerInterface.Delete(context.TODO(), submariner.Name, metav1.DeleteOptions{})).To(Succeed())
+
+		Eventually(func() bool {
+			_, err := submarinerInterface.Get(context.TODO(), submariner.Name, metav1.GetOptions{})
+			return apierrors.IsNotFound(err)
+		}, uninstall.ComponentReadyTimeout*2).Should(BeTrue(), "Submariner resource not deleted")
+
+		By("Submariner resource deleted")
+
+		close(stopCh)
+
+		routeAgentPodMonitor.assertUninstallPodsCompleted()
+		gatewayPodMonitor.assertUninstallPodsCompleted()
+
+		_, err := framework.KubeClients[framework.ClusterA].AppsV1().DaemonSets(framework.TestContext.SubmarinerNamespace).Get(context.TODO(),
+			names.AppendUninstall(names.GatewayComponent), metav1.GetOptions{})
+		Expect(apierrors.IsNotFound(err)).To(BeTrue(), fmt.Sprintf("Unexpected DaemonSet %q found",
+			names.AppendUninstall(names.GatewayComponent)))
+
+		_, err = framework.KubeClients[framework.ClusterA].AppsV1().DaemonSets(framework.TestContext.SubmarinerNamespace).Get(context.TODO(),
+			names.AppendUninstall(names.RouteAgentComponent), metav1.GetOptions{})
+		Expect(apierrors.IsNotFound(err)).To(BeTrue(), fmt.Sprintf("Unexpected DaemonSet %q found",
+			names.AppendUninstall(names.RouteAgentComponent)))
+
+		// Verify the Cluster and Endpoint resources were removed from the broker.
+
+		submarinerClient, err := submarinerclientset.NewForConfig(brokerRestConfig)
+		Expect(err).To(Succeed())
+
+		clusterID := framework.TestContext.ClusterIDs[framework.ClusterA]
+
+		_, err = submarinerClient.SubmarinerV1().Clusters(framework.TestContext.SubmarinerNamespace).Get(context.TODO(),
+			clusterID, metav1.GetOptions{})
+		Expect(apierrors.IsNotFound(err)).To(BeTrue(), fmt.Sprintf("Unexpected Cluster %q found on the broker", clusterID))
```

Comment:
```suggestion
		Expect(apierrors.IsNotFound(err)).To(BeTrue(), fmt.Sprintf("Unexpected Cluster %q found", clusterID))
```

We are validating in the local cluster (not on broker) right?

Reply:
It's actually meant to check the broker NS. We really should check both to be complete.